### ISSUE TITLE
Issue with training

### ISSUE THREAD

**Issue by seovchinnikov** [ildoonet/tf-pose-estimation#132]
what if batch size < 16 => exception ?
https://github.com/ildoonet/tf-pose-estimation/blob/8543abfc00a7ff149f64ce84114a3148a93b6074/src/train.py#L252

**Comment by shreyasrajesh** [comment]
I am having a similar issue. I would like to train with lower batch size. Reducing 16 to a smaller number doesn't seem to be solving the problem. @seovchinnikov Were you able to fix it ?

**Comment by ildoonet** [comment]
@seovchinnikov @shreyasrajesh Yeah that is... something you should change if you want to use small size of batch.

**Comment by lampsonSong** [comment]
@seovchinnikov @shreyasrajesh  
I change the code of 
" feed_dict={q_inp: np.array((sample_image + val_image)*(args.batchsize // 16))} " as 
feed_dict={q_inp: np.array((sample_image + val_image)*(max(1, args.batchsize // 16)))}.

It works.

**Comment by Naykira** [comment]
@lampsonSong Hi, I am trying to implement your approach as I face resource exhausted error for batchsize=16 when training. I replace ` feed_dict={q_inp: np.array((sample_image + val_image) * (args.batchsize // 16))}` with ` feed_dict={q_inp: np.array((sample_image + val_image) * (max(1, args.batchsize // 16)))}` as per your suggestion and set batchsize = 1. But I face `IndexError: index 1 is out of bounds for axis 0 with size 1` error at `sample_image = [enqueuer.last_dp[0][i] for i in range(4)]` now. How did you ensure that the training is correctly done when batchsize is set less than 16. Could you please share your modifications? Thanks!

**Comment by lampsonSong** [comment]
@Naykira Hi, I think the reason is : 
the 

> input_node = tf.placeholder(tf.float32, shape=(args.batchsize, args.input_height, args.input_width, 3), name='image')

You set the batchsize as 1, so the placeholder get size 1. You should set the batchsize as 4 or other proper number less than 16. 
Hope to be helpful.

**Comment by vebitsai** [comment]
sample_image is hard-coded to be 4, and val_image 12. 

placeholder is set to the batch size,  so sample_image + val_image must set to batchsize

I assumed that I can change the sample_image by line 249
"sample_image = [enqueuer.last_dp[0][i] for i in range(12)]" 
so to have a size of Batch size of 24 in my case.
